crates/redeem/src/lib.rs

```diff
@@ -557,12 +638,8 @@ impl<T: Config> Module<T> {
         );
         // NOTE: temporary workaround until we delete
         ensure!(
-            !<RedeemRequests<T>>::get(*redeem_id).completed,
-            Error::<T>::RedeemCompleted
-        );
-        ensure!(
-            !<RedeemRequests<T>>::get(*redeem_id).cancelled,
-            Error::<T>::RedeemCancelled
+            <RedeemRequests<T>>::get(*redeem_id).status == RedeemRequestStatus::Pending,
+            Error::<T>::InvalidRedeemStatus
```

Comment:
I think we probably want to keep the error reason for now, you could add `Into<Error<T>>` for the `RedeemRequestStatus`.